### PR TITLE
Added new smol 🤏 Multimodal RAG System with ColSmolVLM and SmolVLM recipe 🧑‍🍳️

### LINKED ISSUE #243: New smol 🤏 Multimodal RAG System with ColSmolVLM and SmolVLM recipe 🧑‍🍳️
I propose adding a new recipe demonstrating how to build a Multimodal Retrieval-Augmented Generation (RAG) system using a combination of the [ColSmolVLM](https://huggingface.co/vidore/colsmolvlm-alpha) retrieve, and the [SmolVLM](https://huggingface.co/HuggingFaceTB/SmolVLM-Instruct).

This recipe will guide users through the process of setting up and running a multimodal RAG system, optimized for use with consumer GPUs and even with Colab's free-tier.

### PR DESCRIPTION
**Pull request by sergiopaniego**
# What does this PR do?

Fixes #243

## Who can review?

@merveenoyan and @stevhliu

### CODE REVIEW

**Comment by andimarafioti** [comment]
very cool!

**Comment by sergiopaniego** [comment]
> View / edit / reply to [this conversation](https://app.reviewnb.com/huggingface/cookbook/pull/244/discussion/#comment-2520225070) on ReviewNB
> 
> _andimarafioti commented on 2024-12-05T12:45:51Z_ ----------------------------------------------------------------
> 
> Have you tried to also quantize the docs_retrieval_model?

The library currently doesn't support `quantization_config` as in `transformers`:

image

It could be possible but I think it would add a lot of complexity to the notebook without that native [support](https://github.com/AnswerDotAI/byaldi/blob/4583c0732fac06cca5624c53a539a7785d226ce8/byaldi/RAGModel.py#L17).

File: notebooks/en/_toctree.yml [hunk 1]
```diff
@@ -94,6 +94,8 @@
           title: Fine-Tuning a Vision Language Model (Qwen2-VL-7B) with the Hugging Face Ecosystem (TRL)
         - local: multimodal_rag_using_document_retrieval_and_reranker_and_vlms
           title: Multimodal RAG with ColQwen2, Reranker, and Quantized VLMs on Consumer GPUs
+        - local: multimodal_rag_using_document_retrieval_and_smol_vlm
+          title: "Smol Multimodal RAG: Building with ColSmolVLM and SmolVLM on Colab's Free-Tier GPU"
```
Comment (stevhliu) [inline notebooks/en/_toctree.yml:98]:
```suggestion
          title: Smol Multimodal RAG: Building with ColSmolVLM and SmolVLM on Colab's Free-Tier GPU
```

**Review by stevhliu — COMMENTED** [review]
Super nice job!

**Comment by sergiopaniego** [comment]
<p>In the actual notebook, it's not shown twice: <a href="https://github.com/huggingface/cookbook/blob/12a99a3d8c4f5607bb382e1d8536e60392e653b1/notebooks/en/multimodal_rag_using_document_retrieval_and_smol_vlm.ipynb" target="_blank">https://github.com/huggingface/cookbook/blob/12a99a3d8c4f5607bb382e1d8536e60392e653b1/notebooks/en/multimodal_rag_using_document_retrieval_and_smol_vlm.ipynb</a></p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/244/discussion/#comment-2523823760'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by sergiopaniego** [comment]
<p>Added! Comparison of these two systems and the other two multimodal RAG recipes :)</p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/244/discussion/#comment-2523823780'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by sergiopaniego** [comment]
Updated 🤗

**Review by stevhliu — APPROVED** [review]
Awesome! Once your byaldi PR is merged, we can merge this one as well 🤗

File: notebooks/en/_toctree.yml [hunk 2]
```diff
@@ -94,6 +94,8 @@
           title: Fine-Tuning a Vision Language Model (Qwen2-VL-7B) with the Hugging Face Ecosystem (TRL)
         - local: multimodal_rag_using_document_retrieval_and_reranker_and_vlms
           title: Multimodal RAG with ColQwen2, Reranker, and Quantized VLMs on Consumer GPUs
+        - local: multimodal_rag_using_document_retrieval_and_smol_vlm
+          title: Smol Multimodal RAG: Building with ColSmolVLM and SmolVLM on Colab's Free-Tier GPU
```
Comment (stevhliu) [inline notebooks/en/_toctree.yml:98]: Can't have a `:` in the title

```suggestion
          title: Smol Multimodal RAG, Building with ColSmolVLM and SmolVLM on Colab's Free-Tier GPU
```

**Review by merveenoyan — COMMENTED** [review]
left some nits! otherwise looks good :)

**Comment by sergiopaniego** [comment]
Thanks for the suggestions @merveenoyan 😄 Applied!

**Pull request merged**